system.go

```diff
@@ -30,6 +30,14 @@ import (
 	_ "github.com/elastic/go-sysinfo/providers/windows"
 )
 
+type ProviderOption func(*registry.ProviderOptions)
+
+var WithHostFS = func(hostfs string) ProviderOption {
```

Comment:
Make this `func WithHostFS` to avoid having a global variable that could be mutated to change library behavior. Plus I think it will be move discoverable in the godoc as a function.

Please add some godoc to it.